[Feat]챌린지 인증 API

Hminkyung opened the pull request:
## #️⃣ 연관된 이슈

> 관련된 이슈 번호를 적어주세요.\
> Close #51 

## ✨ 작업 내용 (Summary)

> 이번 PR에서 작업한 내용을 간략히 설명해주세요. (이미지 첨부 가능)
챌린지 인증 관련 API들을 전부 구현합니다.

1. 텍스트 인증 API
2. 사진 인증 API 

7. 댓글 작성 API - 저희 회의 했을 때 깊이가 0과 1로만 진행한다고 해서 GROUP을 사용해서 만들지는 않았습니다. 근데 만약에 댓글, 대댓글 말고도 더 추가적으로 진행한다고 하면 수정할게용
8. 댓글/대댓글 조회 API -> 해당 인증글 입력하면 다 볼 수 있어요
9. 댓글 수정 API
10. 댓글 삭제 API -> 이거는 완전히 삭제하면 추후에 막 부모 댓글 없는데 막 사라지고 이런 이슈 발생할까봐 SOFTDELETE로 구현해뒀습당

--여기 아래는 조금 더 수정할게요
(3. 인증글 상세 조회 API -> 이거는 인증글 볼 수 있는 API이고
4. 사용자 본인의 인증글 목록 조회 -> 이거는 디자인이 없는것 같기도 한데 일단 무조건 필요할 것 같아서 만들어뒀습니다. 자신의 작성한 인증글 목록 API입니다.
5. 챌린지랑 라운드별 인증글 조회 -> 해당 챌린지에 들어갔을 때 보이는 인증글들 목록입니다. 근데 라운드가 아직 구현은 안되어있어서 일시적으로 파라미터로 받고 필요한 코드들은 주석처리해서 구현되도록 했습니다.
6. )

---

## ✅ 변경 사항 체크리스트

> 다음 항목들을 확인하고 체크해주세요.

- [ ] 코드에 영향이 있는 모든 부분에 대한 테스트를 작성하고 실행했나요?
- [ ] 문서를 작성하거나 수정했나요? (필요한 경우)
- [ ] 중요한 변경 사항이 팀에 공유되었나요?

---

## 🧪 테스트 결과

> 코드 변경에 대해 테스트를 수행한 결과를 요약해주세요.

* **테스트 환경:** (예: 로컬, 개발 서버 등) 로컬
* **테스트 방법:** (예: Postman, 단위 테스트, 수동 기능 테스트 등) SWAGGER
* **결과 요약:** (예: 모든 테스트 통과, 새로운 테스트 케이스 3개 추가 완료) 모든 테스트 통과

---

## 📸 스크린샷

> 관련된 스크린샷 또는 GIF가 있다면 여기에 첨부해주세요.
<img width="985" height="1077" alt="스크린샷 2025-11-30 184139" src="https://github.com/user-attachments/assets/a5e97f27-de84-482b-9585-d4382d93b4da" />
<img width="1388" height="1328" alt="스크린샷 2025-11-30 191738" src="https://github.com/user-attachments/assets/fe968f3c-a9ed-4277-ae0f-c3294f4e1812" />
<img width="962" height="1102" alt="스크린샷 2025-11-30 195518" src="https://github.com/user-attachments/assets/4cc70d52-7239-4e8b-b0a3-fbf660524c6e" />
<img width="1432" height="1422" alt="스크린샷 2025-11-30 210438" src="https://github.com/user-attachments/assets/bdef6df1-2b3f-470f-a10e-8d85483cdf52" />
<img width="1416" height="1128" alt="스크린샷 2025-11-30 213828" src="https://github.com/user-attachments/assets/a582b6c0-3b04-47b7-9032-f749efefbb8d" />
<img width="1445" height="1219" alt="스크린샷 2025-11-30 214354" src="https://github.com/user-attachments/assets/672623c4-3a0e-4ebc-9c05-de34b1358839" />
<img width="1432" height="1133" alt="스크린샷 2025-11-30 214540" src="https://github.com/user-attachments/assets/e39228ca-e71d-406f-be38-3354b906e488" />
<img width="928" height="872" alt="스크린샷 2025-11-30 214808" src="https://github.com/user-attachments/assets/0aa020e9-42b8-42d7-9035-543817b2638c" />



---

## 💬 리뷰 요구사항

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요.
라운드 부분은 주석처리만 진행한 채로 구현했는데 혹시 빼먹은거나 그 외에 조금 인증 이상하다 구현 이상하다 싶은거 한번만 봐주세요ㅠㅠㅠ 
---

## 📎 참고 자료

> 관련 문서, 레퍼런스 링크 등이 있다면 여기에 첨부해주세요.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **새로운 기능**
  * 인증글에 댓글 CRUD 및 대댓글(익명 옵션, 소프트 삭제, 좋아요 카운트) 추가; 댓글 목록 페이징/정렬 제공  
  * 텍스트/사진 인증글 작성 기능 추가(라운드 연동·카운트 갱신, 사진 업로드 키 지원)  
  * 인증·댓글 응답 포맷 및 관련 DTO/컨버터 추가

* **DB 마이그레이션**
  * 댓글·인증 관련 컬럼·인덱스·외래키를 조건부로 추가하는 마이그레이션 스크립트 추가

* **성공/오류 메시지**
  * 댓글·인증·파일 업로드 관련 성공/오류 코드 및 메시지 확장 추가

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Hminkyung committed:
feat: 인증 관련 API 구현
글 인증 API
사진 인증 API
인증글 상세 조회 API
사용자 본인의 인증글 목록 API
챌린지랑 라운드별 인증글 조회 API

Hminkyung committed:
feat: 댓글과 대댓글 API 구현
댓글 작성 API 구현
대댓글 작성 API 구현
댓글 수정 API 구현
댓글 목록 조회 API 구현
댓글 삭제 API 구현

Hminkyung committed:
merge: develop과 병합 충돌 해결

yc3697 commented on src/main/java/com/hrr/backend/domain/comment/service/CommentServiceImpl.java:
```diff
@@ -0,0 +1,129 @@
+
+    /** 댓글/대댓글 조회 */
+    @Override
+    public CommentListResponseDto getComments(Long verificationId) {
```
soft delete 방식을 적용하고 있으니 댓글 조회 시에도 isDeleted=false 여부를 확인하여 필터링하는 작업이 필요합니다.

yc3697 replied:
그리고 나중에 댓글 조회도 페이지네이션 적용해주시면 좋을 것 같습니다!

yc3697 commented on src/main/java/com/hrr/backend/domain/comment/service/CommentServiceImpl.java:
```diff
@@ -0,0 +1,129 @@
+    /** 댓글 삭제 (Soft Delete) */
+    @Override
+    @Transactional
+    public void deleteComment(Long commentId, Long userId) {
```
soft delete 후 DB에도 저장이 필요합니다.

yc3697 reviewed (changes requested):
1. 리뷰 남긴 거 해결
2. 코드래빗의 리뷰 중 해결 필요
      - 에러 코드 중복 해결
      - CommentResponseDto에서 createdAt LocalDateTime으로 바꾸기
      - 대댓글도 한 번에 가져와서 N+1 문제 해결 추천
      - updateComment 중복 해결
      - Dto에 어노테이션 누락 해결
      - VerificationController 경로 수정
      - roundId 타입 일치
3. UserId는 이미 검증을 거쳐 CustomUserDetails에 저장이 되기 때문에 userId를 별도로 받아온다거나 request에서 token을 추출하여 userId를 뽑는 등의 작업이 불필요합니다. userId를 받아오는 건 다른 사용자의 리소스를 조회하는 api 제외하고는 전부 변경 바랍니다.
4. Controller에서 응답 타입이 ResponseEntity로 되어있는데 프로젝트의 모든 응답 방식이 ApiResponse를 사용하는 것으로 통일되어 있기 때문에 이 또한 변경이 필요합니다.
5. Comment 엔티티의 변경사항을 반영하기 위한 마이그레이션 스크립트 파일 작성이 필요합니다.

파이팅...ㅎㅎ

Hminkyung committed:
merge to develop

Hminkyung committed:
merge error solve

Hminkyung committed:
refactor: 코드래빗 리뷰 반영

Hminkyung committed:
refactor: 코드 리뷰 반영 수정

Hminkyung committed:
feat: sql 파일 추가

Hminkyung committed:
merge to develop

Hminkyung committed:
refactor: 에러 수정

Hminkyung committed:
refactor: 코드 리뷰 전체 반영
CommentResponseDto createdAt LocalDatetime변경
대댓글을 한번에 가져와서 N+1 문제 해결
updateComment 중복 해결
Dto에 어노테이션 누락 해결
VerificationController 경로 수정
roundId 타입 일치
UserId를 받아와서 변경
ApiResponse 응답 방식으로 통일
Comment 엔티티와 Verification 마이그레이션 스크립트 파일 작성
soft delete 적용해서 댓글 조회 시에 필터링
soft delete 후에 DB 저장
댓글 조회 페이지네이션 처리

Hminkyung committed:
feat: S3 연동 작업-글인증과 사진인증

Hminkyung committed:
merge to devleop

Hminkyung committed:
feat: 사진인증 및 글인증 S3이용해서 구현 완료

yc3697 reviewed (approved):
확인했습니다!

댓글 조회할 떄 isDeleted=false 여부 조회
CommentCreateRequestDto에 어노테이션 누락 해결

이거까지만 수정하고 머지해주시면 될 것 같아요 수고하셨습니다~

Hminkyung committed:
refactor:
댓글 조회 시 isDelteted=false 여부 조회
CommentCreateRequestDto 에 어노테이션 누락 해결

Hminkyung committed:
Merge branch 'develop' into feat/51-challenge-verification
